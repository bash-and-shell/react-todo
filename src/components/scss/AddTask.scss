.add-task {

  text-align: left;
  
  &-group{
    display: flex;
    gap: 1rem;
  }
 
}