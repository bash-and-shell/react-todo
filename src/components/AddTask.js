import React from 'react';
import { Button, Form } from 'react-bootstrap';
import './scss/AddTask.scss'

const AddTask = (props) => {
  return (
    <Form className="add-task" onSubmit={props.onSubmit}>
              <Form.Label>Enter New Task</Form.Label>

      <Form.Group controlId="addTask" className='add-task-group'>
        <Form.Control type="text" placeholder="Enter task" />
        <Button className="searchforbutton" variant="primary" type="submit">Add</Button>
      </Form.Group>
    </Form>
  );
}

export default AddTask;
