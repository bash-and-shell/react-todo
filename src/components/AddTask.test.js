import renderer from 'react-test-renderer'
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import AddTask from './AddTask';


test('matches snapshot', () => {
  const component = renderer.create(
    <AddTask enabled={false} onSubmit={jest.fn()}/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('check button enables with typing', () => {
  render(    <AddTask enabled={false} onSubmit={jest.fn()}/>  )

  expect(screen.getByRole('button')).toHaveAttribute('disabled');
  
  fireEvent.change(screen.getByPlaceholderText("Enter task"), { target: {value: 'test'} })

  expect(screen.getByRole('button')).not.toHaveAttribute('disabled');
})

test('expect onSubmit to be called', () => {
  const onSubmit = jest.fn()

  render(<AddTask enabled={false} onSubmit={onSubmit}/>  )
  
  fireEvent.change(screen.getByPlaceholderText("Enter task"), { target: {value: 'test'} })

  expect(screen.getByRole('button')).not.toHaveAttribute('disabled');

  fireEvent.click(screen.getByRole('button'))

  expect(onSubmit).toHaveBeenCalled()

})

test('expect onSubmit to not be called with no typing', () => {
  const onSubmit = jest.fn()

  render(<AddTask enabled={false} onSubmit={onSubmit}/>  )

  fireEvent.click(screen.getByRole('button'))

  expect(onSubmit).not.toHaveBeenCalled()

})