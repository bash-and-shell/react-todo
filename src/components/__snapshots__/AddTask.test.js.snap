// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className="add-task"
  onSubmit={[Function]}
>
  <label
    className="form-label"
  >
    Enter New Task
  </label>
  <div
    className="add-task-group"
  >
    <input
      className="form-control"
      id="addTask"
      onChange={[Function]}
      placeholder="Enter task"
      type="text"
      value=""
    />
    <button
      className="btn btn-primary"
      disabled={true}
      type="submit"
    >
      Add
    </button>
  </div>
</form>
`;
